src/main/java/it/gov/pagopa/payment/service/payment/qrcode/QRCodeConfirmationServiceImpl.java

```diff
@@ -41,13 +43,13 @@ public TransactionResponse confirmPayment(String trxId, String merchantId, Strin
             TransactionInProgress trx = repository.findByIdThrottled(trxId);
 
             if (trx == null) {
-                throw new ClientExceptionNoBody(HttpStatus.NOT_FOUND, "[CONFIRM_PAYMENT] Cannot found transaction having id: " + trxId);
+                throw new TransactionNotFoundOrExpiredException(ExceptionCode.TRX_NOT_FOUND_OR_EXPIRED, "[CONFIRM_PAYMENT] Cannot found transaction having id: " + trxId);
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor